modules/gateway/peersmanager.go

```diff
@@ -60,6 +60,25 @@ func (g *Gateway) numOutboundPeers() (numOutboundPeers int) {
 	return numOutboundPeers
 }
 
+// randomPotentialPeer returns a random node from the gateway. Unlike
+// randomNode, it is not truly random because it prioritizes some nodes over
+// others. An error can be returned if there are no nodes in the node list.
+func (g *Gateway) randomPotentialPeer() (modules.NetAddress, error) {
+	if len(g.nodes) == 0 {
+		return "", errNoPeers
+	}
+
+	// Try three times to select a prioritized node.
+	for i := 0; i < 3; i++ {
+		addr, _ := g.randomNode()
+		if g.nodes[addr].Prioritize {
+			return addr, nil
+		}
+	}
```

Comment:
ok, so something like:
```go
func permanentPeerManager() {
    // try the previous outbound nodes first
    for _, node := range g.nodes {
        if node.WasOutboundPeer {
            tryConnect(node)
        }
    }

    // afterwards, use random selection
    for {
        tryConnect(g.randomPotentialPeer())
        sleep()
    }
}
```
I'm also thinking that `g.randomPotentialPeer` should filter out any addrs that we're already connected to.